stl/inc/xstring

```diff
@@ -4392,8 +4330,10 @@ public:
         // exchange a string in large mode with one in small mode
         const pointer _Ptr = _Starts_large._Bx._Ptr;
         _Destroy_in_place(_Starts_large._Bx._Ptr);
+
         _Starts_large._Activate_SSO_buffer();
-        _Traits::copy(_Starts_large._Bx._Buf, _Starts_small._Bx._Buf, _BUF_SIZE);
+        _Traits::copy(_Starts_large._Bx._Buf, _Starts_small._Bx._Buf, _Starts_small._Mysize + 1);

```

Comment:
Are we intentionally removing the "copy the entire SSO buffer" microoptimization here?

Reply:
ah, this is a change from when SBO was annotated; it should probably be reverted.

Reply:
I can validate this change if you're too busy with non-STL work this week. (I would also like to try reverting https://github.com/microsoft/STL/pull/3164#discussion_r1044943528 .) I hope that wouldn't reset ASAN testing since these are non-ASAN-specific codepaths.

Reply:
I can do so; the LKG stuff took _way_ less time than I expected (it was green without work, surprisingly).